include/lug/System/Exception.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <string>
+#include <lug/System/Export.hpp>
+
+namespace lug {
+namespace System {
+
+class LUG_SYSTEM_API Exception {
+public:
+    Exception(const char *typeName, const std::string &description, const char* file, const char* function, long line);
+    Exception(Exception&& other);
+    Exception(const Exception& other);
+    Exception& operator=(const Exception& other);
+    Exception& operator=(const Exception&& other);
+    ~Exception() throw() {}
+
+    const std::string&              getTypeName() const;
+    const std::string&              getDescription() const;
+    const std::string&              getFile() const;
+    const std::string&              getFunction() const;
+    const long                      getLine() const;
+
+    const char*                     what() const;
```

Comment:
You don't inherit from std::exception, so for what reason do you don't return a std::string to avoid the usage of _fullDesc?